pkg/xds/server/dataplane_metadata_tracker.go

```diff
@@ -71,3 +96,25 @@ func (d *DataplaneMetadataTracker) OnFetchRequest(context.Context, *envoy.Discov
 
 func (d *DataplaneMetadataTracker) OnFetchResponse(*envoy.DiscoveryRequest, *envoy.DiscoveryResponse) {
 }
+
+func (d *DataplaneMetadataTracker) registerDataplane(dp *core_mesh.DataplaneResource) error {
+	if d.environmentType != core.UniversalEnvironment {
+		return nil
+	}
+	existing := &core_mesh.DataplaneResource{}
+	err := d.rm.Get(context.Background(), existing, store.GetBy(model.MetaToResourceKey(dp.GetMeta())))
+	if err == nil {
+		return errors.Errorf("provided Dataplane %s already exists in %s mesh", dp.GetMeta().GetName(), dp.GetMeta().GetMesh())
+	}
+	if !store.IsResourceNotFound(err) {
+		return err
+	}
+	return d.rm.Create(context.Background(), dp, store.CreateBy(model.MetaToResourceKey(dp.GetMeta())))
+}
+
+func (d *DataplaneMetadataTracker) unregisterDataplane(dp *core_mesh.DataplaneResource) error {
+	if d.environmentType != core.UniversalEnvironment {
+		return nil
+	}
+	return d.rm.Delete(context.Background(), &core_mesh.DataplaneResource{}, store.DeleteBy(model.MetaToResourceKey(dp.GetMeta())))
+}
```

Comment:
Ok, I don't like the idea of putting those changes in `dataplane_metadata_tracker`. Dataplane Metadata Tracker is responsible for well... tracking metadata. IMHO we should create Dataplane Lifecycle callbacks that we put right after Dataplane Metadata Tracker in callback chain. Dataplane Lifecycle would then use Metadata extracted by a tracker and create and delete DP if necessary.

Additionally, I think we should only delete dataplanes that we created so instead of if on the environment there should be an if whether we created DP using Dataplane Lifecycle